### PR TITLE
Work during March and late February

### PR DESCRIPTION
**Pull request by BenMueller1**
- added customizableModel
- added notebook to perform hyperparameter sweep using the Optuna library
- added new mean adjustment functions
- added detailed logging for all models
- added notebooks to inspect and plot results of data generation functions
- various other small changes & bugfixes

### CODE REVIEW

File: yeastdnnexplorer/data_loaders/synthetic_data_loader.py [hunk 1]
```diff
@@ -136,9 +159,33 @@ def prepare_data(self) -> None:
...
+                    max_mean_adjustment=self.max_mean_adjustment,
+                )
+                for tf_index in range(sum(self.n_sample))
+            ]
+
+        # take absolute value, this is all we care about
         perturbation_effects_list = [
-            generate_perturbation_effects(binding_data_tensor, tf_index=tf_index)
-            for tf_index in range(sum(self.n_sample))
+            torch.abs(perturbation_effects)
```
Comment (cmatKhan) [inline yeastdnnexplorer/data_loaders/synthetic_data_loader.py:187]: I'm concerned that doing this here is possibly affecting what we're expecting to see downstream

**Review by cmatKhan — COMMENTED** [review]
Please just check that taking the absolute value in the data loader step is the right place for it -- it may be

**Comment by cmatKhan** [comment]
remind me what the issue has been with the python version? why is the testing failing on ubuntu 20?

File: yeastdnnexplorer/data_loaders/synthetic_data_loader.py (hunk 1 above)
Comment (BenMueller1) [reply yeastdnnexplorer/data_loaders/synthetic_data_loader.py:187]: Should we be doing this in the functions in generate_data.py instead?

**Comment by BenMueller1** [comment]
> remind me what the issue has been with the python version? why is the testing failing on ubuntu 20?

I believe that it was because I wasn't able to run the precommit locally without updating the python version in pyproject.toml

File: yeastdnnexplorer/data_loaders/synthetic_data_loader.py (hunk 1 above)
Comment (cmatKhan) [reply yeastdnnexplorer/data_loaders/synthetic_data_loader.py:187]: no, it the generate_data functions should generate positive and negative effect sizes.

I think this is fine here, but it is confusing to me why when we looked at results, if this is already happening, that it seemed like there were positive and negative effects. If we were looking at data that was coming out of this, the we should only be seeing positive effects

**Pull request closed without merging**